### ISSUE TITLE
Investigate reuse strategies for docker image

### ISSUE THREAD

**Issue by ckunki** [exasol/ai-lab#59]
### Potential use cases

UC-1
* notebook developer Nadine works on creating a new Jupyter notebook or updating an existing one 
* Nadine wants to use new libraries that are not available yet in the latest release on docker-hub
* Nadine therefor wants to build a private Docker image from the branch Nadine is currently working on 

UC-2
* As image creation currently (Nov 2023) takes around 7 minutes Nadine wants to reuse the image in follow-up usage 

UC-3
* Nadine changed a file or dependency that requires to re-create the Docker image taking the change into account

**Comment by ckunki** [comment]
Using a stripped playbook I measured a total duration of 35 seconds. 
The main runtime is used for copying ~ 1MB files into the Docker container.

**Comment by ckunki** [comment]
Comments from @tkilias 
* Installation of poetry takes quite long and I think we don't need poetry
* Yes, APT and PIP installs take a while but there are only limited things we could do about it 
* In a Docker Container we don't need systemd and motd setup as they probably won't work anyway

**Comment by ckunki** [comment]
See https://stackoverflow.com/questions/27985334/why-is-copying-a-directory-with-ansible-so-slow:
TLDR: use `synchronize` instead of `copy`.

**Comment by ckunki** [comment]
See also https://docs.pytest.org/en/7.4.x/how-to/capture-stdout-stderr.html
```python
def test_disabling_capturing(capsys):
    print("this output is captured")
    with capsys.disabled():
        print("output not captured, going directly to sys.stdout")
    print("this output is also captured")

```

**Comment by ckunki** [comment]
I used 

```yaml
- name: Copy notebook content
  ansible.builtin.synchronize:
    src: "roles/jupyter/files/notebook/"
    dest: /root/notebooks
    rsync_opts:
      - "--chmod=0644"
```

And got error message 
> protocol version mismatch -- is your shell clean?
> (see the rsync man page for an explanation)
> rsyncerror: protocol incompatibility (code 2) at compat.c(178) [sender=3.1.3]

**Comment by ckunki** [comment]
Capturing stdout only works then pytest is called with `-o log_cli=true -o log_cli_level=INFO`.
But when these cli options are provided capturing is not required anymore, as pytest will log ansible output anyway.

**Comment by ckunki** [comment]
For reusing a Docker Image in pytest DSS added support for CLI option `--dss-docker-image` in https://github.com/exasol/data-science-sandbox/issues/69
